plugins/arcanist/arcanist.plugin.zsh

```diff
@@ -13,10 +13,26 @@ alias ardc='arc diff --create'
 alias ardnu='arc diff --nounit'
 alias ardnupc='arc diff --nounit --plan-changes'
 alias ardpc='arc diff --plan-changes'
+alias ardcp= 'arc diff --preview' # creates a new diff in the phab interface
 
 alias are='arc export'
 alias arh='arc help'
 alias arl='arc land'
 alias arli='arc lint'
 alias arls='arc list'
-alias arpa='arc patch'
+
+#
+# Functions
+# (sorted alphabetically)
+#
+
+ardu() {
+# With this, both, `ardu https://arcanist-url.com/<REVISION>`, and `ardu <REVISION>` work.
+arc diff --update "$(echo "$1" | awk '{n=split($1,a,"/");print a[n]}')"
```

Comment:
You can use `${(s./.)var}` to split by `/` a variable, then `${var[-1]}` to get the latest element of the var array. All together:
```suggestion
arc diff --update "${${(s./.)1}[-1]}"
```
Don't forget to indent as well

Reply:
The suggested update will not work when you just want to provide the revision id. i.e.,

`ardu http://example.com/D33333` will work

`D33333` won't work

I suggest using awk for readability and the above point. 

Another point for not using awk would be that awk may not be available, however, using rip grep, I can see it being used widely, e.g., the ubuntu plugin in line 88, the gradle plugin in line 124 among others.

Reply:
We can make it a two-part algorithm then:
```suggestion
local -a URL=("${(s:/:)1}")
arc diff --update "${URL[-1]}"
```

Reply:
Thanks for the suggestion, this one works great 👍 . I am extracting this into a shared function to prevent code duplication.

Test result
![image](https://user-images.githubusercontent.com/13885630/75626745-67ba3780-5bca-11ea-997d-f5bfcdce890a.png)